src/decode.ts

```diff
@@ -0,0 +1,83 @@
+import { Uint8ArrayList } from 'uint8arraylist'
+import errcode from 'err-code'
+import { FrameHeader, FrameType, HEADER_LENGTH, YAMUX_VERSION } from './frame.js'
+import { ERR_DECODE_INVALID_VERSION } from './constants.js'
+import type { Source } from 'it-stream-types'
+
```

Comment:
it'd be great if we have some unit tests for file (could be in a separate issue/PR)